src/main/java/dev/blaauwendraad/masker/json/config/KeyMaskingConfig.java

```diff
@@ -0,0 +1,377 @@
+        }
+
+        /**
+         * Mask all boolean values with the provided value.
+         * For example, {@literal "maskMe": true -> "maskMe": "&&&"}.
+         * <p>
+         * Masking booleans with {@literal '&&&'} is the default behaviour if no boolean masking option is set.
+         *
+         * @see #disableBooleanMasking()
+         * @see #maskBooleansWith(boolean)
+         *
+         * @return the builder instance
+         */
+        public Builder maskBooleansWith(String value) {
+            if (maskBooleansWith != null) {
+                throw new IllegalStateException("'maskBooleansWith(String)' was already set");
+            }
+            checkMutuallyExclusiveBooleanMaskingOptions();
+            maskBooleansWithString = Objects.requireNonNull(value);
+            return this;
+        }
+
+        /**
+         * Mask all boolean values with the provided value.
+         * For example, "maskMe": true -> "maskMe": false.
+         *
+         * @see #disableBooleanMasking()
+         * @see #maskBooleansWith(String)
+         *
+         * @return the builder instance
+         */
+        public Builder maskBooleansWith(boolean value) {
+            if (maskBooleansWith != null) {
+                throw new IllegalStateException("'maskBooleansWith(boolean)' was already set");
+            }
+            checkMutuallyExclusiveBooleanMaskingOptions();
+            maskBooleansWith = value;
+            return this;
+        }
+
+        /**
+         * Builds the {@link KeyMaskingConfig} instance.
+         *
+         * @return the {@link KeyMaskingConfig} instance
+         */
+        public KeyMaskingConfig build() {
+            if (maskStringsWith == null && maskStringCharactersWith == null) {
+                maskStringsWith = "***";
+            }
+            if (disableBooleanMasking == null && maskNumbersWithString == null && maskNumbersWith == null && maskNumberDigitsWith == null) {
+                maskNumbersWithString = "###";
+            }
+            if (disableBooleanMasking == null && maskBooleansWithString == null && maskBooleansWith == null) {
+                maskBooleansWithString = "&&&";
+            }
+            return new KeyMaskingConfig(
+                    maskStringsWith,
+                    maskStringCharactersWith,
+                    disableNumberMasking != null && disableNumberMasking,
+                    maskNumbersWithString,
+                    maskNumbersWith,
+                    maskNumberDigitsWith,
+                    disableBooleanMasking != null && disableBooleanMasking,
+                    maskBooleansWithString,
+                    maskBooleansWith);
+        }
+
+        private void checkMutuallyExclusiveStringMaskingOptions() {
+            if (maskStringsWith != null || maskStringCharactersWith != null) {
+                throw new IllegalStateException("'maskStringsWith(String)' and 'maskStringCharactersWith(String)' are mutually exclusive");
```

Comment:
Should it be an IllegalArgumentException as from the client perspective they supplied the invalid key masking config for the key definition? Or is it conventional in builders to throw an illegal state on misconfiguration?

Reply:
Should be IllegalArgumentException